### PR TITLE
chore: add cleanup option on providers (plumbing)

### LINKED ISSUE #5015: add clean API for a provider
### Is your enhancement related to a problem? Please describe

a provider could register a clean method that would wipe data, 

### Describe the solution you'd like

for example, for podman it could remove some folders, files, etc. to come back in a 'clean state' before creating a new machine

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by benoitf**
### What does this PR do?
extensions can register some cleanup methods

### What issues does this PR fix or reference?

fixes https://github.com/containers/podman-desktop/issues/5015

### How to test this PR?

you can call `registerCleanup` on a provider (but it won't be used)

### CODE REVIEW

File: packages/main/src/plugin/api/provider-info.ts [hunk 1]
```diff
@@ -115,3 +119,10 @@ export interface PreflightCheckEvent {
   type: 'start' | 'stop';
   status: CheckStatus;
 }
+
+export interface ProviderCleanupActionInfo {
```
Comment (jeffmaury) [inline packages/main/src/plugin/api/provider-info.ts:123]: I don't see where it's been used

Comment (benoitf) [reply packages/main/src/plugin/api/provider-info.ts:123]: I was assuming UI will require to display details about each action but as for now nothing is shown yes it's not used
I can remove it and add it later

Comment (benoitf) [reply packages/main/src/plugin/api/provider-info.ts:123]: removed

Comment (benoitf) [reply packages/main/src/plugin/api/provider-info.ts:123]: so it looks like it's used in provider-registry.ts

put it back

Comment (benoitf) [reply packages/main/src/plugin/api/provider-info.ts:123]: it is used there: https://github.com/containers/podman-desktop/blob/f49d5c7d15e805aa8bf8a79562936aa5a66ef446/packages/main/src/plugin/provider-registry.ts#L493-L521

Comment (benoitf) [reply packages/main/src/plugin/api/provider-info.ts:123]: ping @jeffmaury

**Comment by benoitf** [comment]
@jeffmaury you marked conversation as resolved, but is it approved or not ? ;-)

**Review by jeffmaury — APPROVED** [review]
LGTM

**Comment by benoitf** [comment]
merging, all PR checks are ✅

**Pull request merged**